definitions/npm/ramda_v0.x.x/flow_v0.39.x-/ramda_v0.x.x.js

```diff
@@ -699,12 +708,13 @@ declare module ramda {
   // TODO partial
   // TODO partialRight
   // TODO pipeK
-  // TODO pipeP
 
   declare function tap<T>(fn: (x: T) => any, ...rest: Array<void>): (x: T) => T;
   declare function tap<T>(fn: (x: T) => any, x: T): T;
 
-  // TODO tryCatch
+  declare function tryCatch<A, B, E>(tryer: (a: A, ...rest: Array<any>) => B, catcher: (e: E, a: A, ...rest: Array<any>) => B, a: A): B;
+  declare function tryCatch<A, B, E>(tryer: (a: A, ...rest: Array<any>) => B, catcher: (e: E, a: A, ...rest: Array<any>) => B): (a: A) => B;
+  declare function tryCatch<A, B, E>(tryer: (a: A, ...rest: Array<any>) => B): (((catcher: (e: E, a: A, ...rest: Array<any>) => B, a: A) => B) & ((catcher: (e: E, a: A, ...rest: Array<any>) => B) => (a: A) => B));
```

Comment:
I spend so much time to figure this out. I originally had 4 `declare function`, but test `tc5` or `tc6` will break. Only the 3rd declaration will be used.